### ISSUE TITLE
Getting the list of releasesTypeError: Cannot read property 'status' of undefined

### ISSUE THREAD

**Issue by pradgnyak** [github-tools/github-release-notes#120]
Running:

```
gren release
```

Got the following error:
```
🤖  - Generate release notes:
===================================
⠋ Getting the list of releasesTypeError: Cannot read property 'status' of undefined
    at handler (/usr/local/lib/node_modules/github-release-notes/node_modules/github-api/dist/components/Requestable.js:366:40)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
```

**Comment by alexcanessa** [comment]
Hello @pradgnyak,

Please make sure that:

- The token that you've generated has `repo` scope.
- Your user has access to the repo.
- You have internet connectivity when running the command.

See if you still have the issue and let us know :)

**Comment by pradgnyak** [comment]
I have full access to the repo and token was generated has repo scope.
tokenname — admin:org, admin:org_hook, admin:public_key, admin:repo_hook, gist, notifications, repo, user

I have it stored in bash_profile. 
I have internet connectivity. - (made sure).
I am still unable to use the command and getting same error.

Note: My token was generated before installing the package.  Do I need to regenerate the token? Ideally it should not be a requirement.

**Comment by alexcanessa** [comment]
@pradgnyak you're right: it doesn't matter when the token has been generated.

Unluckily the error is coming from `github-api` package that isn't checking for the `Request` to be defined. So to understand it, we have to debug the package itself.

A couple of more question while we're investigating:

- What is the OS?
- Is it a private or a public repo?
- Are you running the command from a `VM`?

**Comment by pradgnyak** [comment]
OS- macOS Sierra version 10.12.6
This is a private repo
No I am not running the command from VM.

This utility looks really helpful, I really like the ease to generate release notes and changelings. I would really appreciate if you could help debug this.

**Comment by alexcanessa** [comment]
@pradgnyak Thanks! Let's try and fix this then.

We're basically running in the same environment then 🤔 

I know it is a bit tricky, but (since it is an external package) you should try this for me:

1. Open `/usr/local/lib/node_modules/github-release-notes/node_modules/github-api/dist/components/Requestable.js`
2. Go to line `364` and place a wonderful `console.log('object', object)` just after `if (object.hasOwnProperty('config')) {`
3. Paste the result here.

And let's see what happens.

**Comment by pradgnyak** [comment]
🤖  - Generate release notes:
===================================
⠋ Getting the list of releasesobject { Error: connect ECONNREFUSED 127.0.0.1:80
    at Object._errnoException (util.js:1031:13)
    at _exceptionWithHostPort (util.js:1052:20)
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1195:14)
  errno: 'ECONNREFUSED',
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '127.0.0.1',
  port: 80,
  config: 
   { adapter: [Function: httpAdapter],
     transformRequest: { '0': [Function: transformRequest] },
     transformResponse: { '0': [Function: transformResponse] },
     timeout: 0,
     xsrfCookieName: 'XSRF-TOKEN',
     xsrfHeaderName: 'X-XSRF-TOKEN',
     maxContentLength: -1,
     validateStatus: [Function: validateStatus],
     headers: 
      { Accept: 'application/vnd.github.v3+json',
        'Content-Type': 'application/json;charset=UTF-8',
        Authorization: 'token XXXXXXXXXXXXXXXXXXXXXXX',
        'User-Agent': 'axios/0.15.3' },
     method: 'GET',
     url: '/repos//releases',
     params: { per_page: 30, page: 1 },
     data: undefined,
     responseType: 'json' },
  response: undefined }
TypeError: Cannot read property 'status' of undefined
    at handler (/usr/local/lib/node_modules/github-release-notes/node_modules/github-api/dist/components/Requestable.js:367:40)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)

**Comment by alexcanessa** [comment]
@pradgnyak just to confirm, did you remove your repo informations from `url`?

```
url: '/repos//releases',
```

**Comment by pradgnyak** [comment]
Nope

**Comment by alexcanessa** [comment]
Interesting. I think that's our clue then.

Are you running the command from your repo folder, right?

If yes, could you force `gren` to use your repo and run

```
gren release --username=[repo_owner] --repo=[repo_name]
```

**Comment by alexcanessa** [comment]
@pradgnyak could you also try and run this from your repo folder and see what it returns?

```
git config remote.origin.url
```

**Comment by pradgnyak** [comment]
Hi I was able to get the command running atlas I am not getting the  error.
But even though I have all the tags and when I run git command
git pull --tag
* [new branch]      sprint1    -> upstream/sprint1
 * [new tag]         1.0.0      -> 1.0.0
 * [new tag]         v1.0.0     -> v1.0.0
 * [new tag]         v1.0.1     -> v1.0.1
Already up-to-date.

Where as when I am running green command
gren release --username=xxxx --repo=yyy results.
✔ Releases found: 0
⠴ Getting tagsLooks like you have no tags! Tag a commit first and then run gren again

**Comment by pradgnyak** [comment]
git config remote.origin.url results in 
https://github.com/pradgnyak/xxxxx.git

**Comment by pradgnyak** [comment]
Okay I am getting the error again when I try to use command 
gren changelog --override username=pradgnyak repo_name=xxxx
Getting the list of releasesobject { Error: connect ECONNREFUSED 127.0.0.1:80
    at Object._errnoException (util.js:1031:13)
    at _exceptionWithHostPort (util.js:1052:20)
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1195:14)
  errno: 'ECONNREFUSED',
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '127.0.0.1',
  port: 80,
  config: 
   { adapter: [Function: httpAdapter],
     transformRequest: { '0': [Function: transformRequest] },
     transformResponse: { '0': [Function: transformResponse] },
     timeout: 0,
     xsrfCookieName: 'XSRF-TOKEN',
     xsrfHeaderName: 'X-XSRF-TOKEN',
     maxContentLength: -1,
     validateStatus: [Function: validateStatus],
     headers: 
      { Accept: 'application/vnd.github.v3+json',
        'Content-Type': 'application/json;charset=UTF-8',
        Authorization: 'token yyyyy',
        'User-Agent': 'axios/0.15.3' },
     method: 'GET',
     url: '/repos//releases',
     params: { per_page: 30, page: 1 },
     data: undefined,
     responseType: 'json' },
  response: undefined }
TypeError: Cannot read property 'status' of undefined
    at handler (/usr/local/lib/node_modules/github-release-notes/node_modules/github-api/dist/components/Requestable.js:367:40)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
⠋ Getting the list of releases^Z

**Comment by alexcanessa** [comment]
Ok @pradgnyak !

The first issue I've found is that `gren` doesn't get the `HTTPS` configuration of the repo, and I've created an issue for this (#122). Until this issue gets fixed, you can run `gren` with the `username` and `repo` options as shown above.

_Note: for the changelog command, looks like you've missed the `--` before the options, otherwise, I suspect you would receive the same error._

The second issue (no tags) is weird, I've tested the repo over `HTTPS` configuration, and (other than the username/repo error) works fine.

What's your node/npm version?

**Comment by alexcanessa** [comment]
@pradgnyak could you please try this? I would like to understand if this 

```
curl -X GET \
  https://api.github.com/repos/:owner/:repo/tags \
  -H 'accept: application/json' \
  -H 'authorization: token xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx' \
  -H 'content-type: application/json'
```

And see if it returns your tags?

**Comment by pradgnyak** [comment]
Hi - I am getting no list of tags
1. git tag 
lists the tags
2. the curl call returns empty list
curl -X GET   https://api.github.com/repos/:user/:repo/tags   -H 'accept: application/json'   -H 'authorization: token xxxxxxxxxxxxxxxxxxxxxxxxxxxxxx'   -H 'content-type: application/json'
[

]

3. But what is weird is that I have another repo where the curl call returns the list.
[
  {
    "name": "v0.1.109",
    "zipball_url": "https://api.github.com/repos/:user/utils/zipball/v0.1.109",
    "tarball_url": "https://api.github.com/repos/:user/utils/tarball/v0.1.109",
    "commit": {
      "sha": "sha",
      "url": "https://api.github.com/repos/pradgnyak/:user/commits/vxvxvvvvv"
    }
  },
  {
    "name": "v0.1.108",
    "zipball_url": "https://api.github.com/repos/:user/utils/zipball/v0.1.108",
    "tarball_url": "https://api.github.com/repos/:user/utils/tarball/v0.1.108",
    "commit": {
      "sha": "fceb7550f96ff66a25b0e204fcabf968220047b9",
      "url": "https://api.github.com/repos/pradgnyak/utils/commits/xvfghr"
    }
  }
]

**Comment by pradgnyak** [comment]
Only difference - in the 2 different repos - is the way tags are created.
The one I am trying to test - tags are created on github
The one that returns the list of tags using the curl command was created as a part of post build action.

**Comment by pradgnyak** [comment]
node - v9.2.0
npm - 5.5.1

**Comment by alexcanessa** [comment]
@pradgnyak ok,

The call that we just tried is just using `GitHub` APIs directly, so at this point, we can at least remove `gren` itself as the source of the problem.

For what concerns the original issue (the `TypeError - status of undefined` thing), your problem has been fixed and you can run `gren release` from the repo folder without specifying owner and repo. 
Use `github-release-notes@0.14.1` (latest). 

For the other problem, it might be easier to follow up on gitter: https://gitter.im/github-release-notes/Lobby

**Comment by pradgnyak** [comment]
Hi I have been able to access the tags. 
1. Created tags using git command and that way it works. Using github release tab as mentioned in your notes didn't.

I am able to generate the changelog but it seems empty.

The changelog as you described should be the commit message is that right?
I am getting
Command -
gren changelog --generate --override --changelog-filename=RELEASE_NOTES.md --username=pradgnyak --repo=jirasd_api

Results in 

# Changelog

## v1.0.4 (08/12/2017)
*No changelog for this release.*

**Comment by alexcanessa** [comment]
1) The default option for `data-source` is issues, e.g. the issues closed between two **git tags**. If you want to show commits instead, you have to use `--data-source=commits` [see more details here](https://github-tools.github.io/github-release-notes/options.html#data-source)
2) `gren release` gets those tags and creates the release notes on the release tab.
3) `gren` last version now gets information for `https` repos too, so you don't have to specify git info anymore.

So, basically, run:

```bash
$ npm i -g github-release-notes
$ gren changelog --override --generate --data-source=commits --changelog-filename=RELEASE_NOTES.md
```

Keep in mind that, if you use a [configuration file](https://github-tools.github.io/github-release-notes/options.html#configuration-file) you can also customise the output.

**Comment by alexcanessa** [comment]
Closing.. if you need further help, feel free to reopen or create a new issue :)
